File: RetailBackend/RetailBackend/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RetailBackend.Models
{
    public class Product
    {
        [Key]
        public int ProductId { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public decimal Price { get; set; }

        public int CategoryId { get; set; }

        [ForeignKey("CategoryId")]
        public Category Category { get; set; }

        public int SellerId { get; set; }

        [ForeignKey("SellerId")]
        public User Seller { get; set; }

        public int StockQuantity { get; set; }

        public bool IsAvailable { get; set; } = true;

        public string ImageUrl { get; set; }
    }
}

File: retail-frontend/src/app/mock-data/product.ts
import { Product } from '../models/product';

export const PRODUCTS: Product[] = [
  {
    productId: 1,
    name: 'Margherita Pizza',
    description: 'Classic cheese pizza',
    price: 250,
    categoryId: 1,
    categoryName: 'Pizza',
    sellerId: 101,
    sellerName: 'Pizza Hub',
    stockQuantity: 10,
    isAvailable: true,
    imageUrl: 'https://www.youngurbanproject.com/wp-content/uploads/2025/03/Product-Marketing.jpg'
  },
  {
    productId: 2,
    name: 'Veg Burger',
    description: 'Crispy veg burger',
    price: 120,
    categoryId: 2,
    categoryName: 'Burger',
    sellerId: 102,
    sellerName: 'Burger Point',
    stockQuantity: 5,
    isAvailable: true,
    imageUrl: 'https://www.youngurbanproject.com/wp-content/uploads/2025/03/Product-Marketing.jpg'
  }
];